superset-frontend/src/SqlLab/components/ResultSet/index.tsx

```diff
@@ -121,115 +113,96 @@ const LimitMessage = styled.span`
+  useEffect(() => {
     if (
-      this.props.cache &&
-      !nextProps.query.cached &&
-      nextProps.query.results &&
-      nextProps.query.results.data &&
-      nextProps.query.results.data.length > 0
+      cache &&
+      query.cached &&
+      query?.results?.data &&
+      query.results.data.length > 0
     ) {
-      this.setState({ data: nextProps.query.results.data }, () =>
-        this.clearQueryResults(nextProps.query),
-      );
+      setCachedData(query.results.data);
+      clearQueryResults(query);
     }
     if (
-      nextProps.query.resultsKey &&
-      nextProps.query.resultsKey !== this.props.query.resultsKey
+      query.resultsKey &&
+      prevQuery?.resultsKey &&
+      query.resultsKey !== prevQuery.resultsKey
     ) {
-      this.fetchResults(nextProps.query);
+      fetchResults(query);
     }
-  }
+  }, [query, cache]);
 
-  calculateAlertRefHeight = (alertElement: HTMLElement | null) => {
+  const calculateAlertRefHeight = (alertElement: HTMLElement | null) => {
     if (alertElement) {
-      this.setState({ alertIsOpen: true });
+      setAlertIsOpen(true);
     } else {
-      this.setState({ alertIsOpen: false });
+      setAlertIsOpen(false);
     }
   };
 
-  clearQueryResults(query: QueryResponse) {
-    this.props.actions.clearQueryResults(query);
-  }
+  const clearQueryResults = (query: QueryResponse) => {
+    actions.clearQueryResults(query);
+  };
 
-  popSelectStar(tempSchema: string | null, tempTable: string) {
+  const popSelectStar = (tempSchema: string | null, tempTable: string) => {
     const qe = {
       id: shortid.generate(),
       name: tempTable,
       autorun: false,
-      dbId: this.props.query.dbId,
+      dbId: query.dbId,
       sql: `SELECT * FROM ${tempSchema ? `${tempSchema}.` : ''}${tempTable}`,
     };
-    this.props.actions.addQueryEditor(qe);
-  }
-
-  toggleExploreResultsButton() {
-    this.setState(prevState => ({
-      showExploreResultsButton: !prevState.showExploreResultsButton,
-    }));
-  }
+    actions.addQueryEditor(qe);
+  };
 
-  changeSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    this.setState({ searchText: event.target.value });
-  }
+  const changeSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(event.target.value);
+  };
 
-  fetchResults(query: QueryResponse) {
-    this.props.actions.fetchQueryResults(query, this.props.displayLimit);
-  }
+  const fetchResults = (query: QueryResponse) => {
+    actions.fetchQueryResults(query, displayLimit);
+  };
 
-  reFetchQueryResults(query: QueryResponse) {
-    this.props.actions.reFetchQueryResults(query);
-  }
+  const reFetchQueryResults = (query: QueryResponse) => {
+    actions.reFetchQueryResults(query);
+  };
 
-  reRunQueryIfSessionTimeoutErrorOnMount() {
-    const { query } = this.props;
+  const reRunQueryIfSessionTimeoutErrorOnMount = () => {
     if (
       query.errorMessage &&
       query.errorMessage.indexOf('session timed out') > 0
```

Comment:
```suggestion
      query?.errorMessage?.indexOf('session timed out') > 0
```
These can be combined with optional chaining

Reply:
![Screenshot 2022-08-25 at 06 21 06](https://user-images.githubusercontent.com/29536522/186567040-146aa1e3-8a98-44cf-99b0-457c919cd45c.png)

Reply:
Oh I see! This is fine as is, in that case.